### ISSUE TITLE
How to skip smallest layers in a pyramidal TIFF ?

### ISSUE THREAD

**Issue by gvincke** [libvips/libvips#1188]
Hi.
Same question as in issue #1 but for an other reason :
I try to create a PTIFF from an image of 540 000 x 4 000 pixels (so a big but "thin" image), and with tiles of 256x256.
I use this command in my terminal (Debian) to do so :
`vips tiffsave --vips-progress --vips-leak input.v output.tiff --tile --pyramid --compression jpeg -Q 95 --tile-width 256 --tile-height 256 --bigtiff`
This command works very well for large ranges of images, but for this one it does not, and just say :
`VipsRegion: valid clipped to nothing`
As i understood it's because the ratio does not allow to create the last tile of 256 pixels width and less than 2 pixels height. So due to this ratio, which only apply to the summit of the pyramid, vips refuse to create my PTIFF. 
So my question is : As my interest focus mainly on the biggest layers, is there a manner to specify that we want vips to create the PTIFF and ignore the last, or n lasts layers ? 
Because here i know that vips is able to do quite a pyramid, but it refuse it just because of the last tile, wich i don't care.
Thanks a lot for your answer, and for your work on vips in general.
Have a nice day.
Grégoire

**Comment by jcupitt** [comment]
Hello @gvincke,

I remember thinking about this when writing the code and deciding that no one would ever want an  image with an aspect ratio that huge, so it would be fine. Ah well!

The tiff pyr saver currently stops when the longest axis of the image fits within one tile. How about changing it so that it stops when the shortest axis fits within a tile? Your smallest image would then be a x16 shrink, so 33750 x 250 pixels (132 x 1 tiles).

@ruven, do you have any thoughts on this? Would a change like that break iipimage?

**Comment by ruven** [comment]
> @ruven, do you have any thoughts on this? Would a change like that break iipimage?

A width of 33750 pixels is still huge, so you wouldn't be able to zoom out using tiles any further than that. IIPImage should still be able to export resized images using an IIIF or CVT command, however. 

Why not just make vips create all possible layers until it's no longer physically possible? Down to a size of 1px, for example. By the way, why is the minimum height in fact 2px and not 1px?

**Comment by jcupitt** [comment]
Good point Ruven, OK, here's a branch with this change if anyone would like to test it. I see:

```
$ vips black x.v 54000 200
$ vips tiffsave x.v x.tif --pyramid --tile
wtiff_layer_new: sub = 1, width = 54000, height = 200
wtiff_layer_new: sub = 2, width = 27000, height = 100
wtiff_layer_new: sub = 4, width = 13500, height = 50
wtiff_layer_new: sub = 8, width = 6750, height = 25
wtiff_layer_new: sub = 16, width = 3375, height = 12
wtiff_layer_new: sub = 32, width = 1687, height = 6
wtiff_layer_new: sub = 64, width = 843, height = 3
wtiff_layer_new: sub = 128, width = 421, height = 1
```

And it seems to work.

**Comment by gvincke** [comment]
@jcupitt and @ruven, thanks for your contributions to the debate, and for the solutions.

@jcupitt : i come back to give you a feedback as soon as possible !

Have a nice day !

**Comment by gvincke** [comment]
Hi,
After some testing, i can confirm that it works very well !
We succeed to save pyramidal tiff event ten times larger that in my first example, with the same height.
Thanks a lot for this improvement !
Have a nice day.
Grégoire

**Comment by jcupitt** [comment]
Thanks for testing! OK, merged to master, this will be in libvips 8.8.
